src/WebJobs.Extensions.DurableTask/ContextImplementations/DurableEntityContext.cs

```diff
@@ -415,12 +417,94 @@ string IDurableEntityContext.StartNewOrchestration(string functionName, object i
+                {
+                    Entity.Current.DeleteState();
+                    return;
+                }
+                else
+                {
+                    throw new InvalidOperationException($"No operation named '{context.OperationName}' was found.");
+                }
+            }
+
+            // check that the number of arguments is zero or one
+            ParameterInfo[] parameters = method.GetParameters();
+            if (parameters.Length > 1)
+            {
+                throw new InvalidOperationException("Only a single argument can be used for operation input.");
+            }
+
+            object[] args;
+            if (parameters.Length == 1)
+            {
+                // determine the expected type of the operation input and deserialize
+                Type inputType = method.GetParameters()[0].ParameterType;
+                object input = context.GetInput(inputType);
+                args = new object[1] { input };
+            }
+            else
+            {
+                args = Array.Empty<object>();
+            }
+
+#if !FUNCTIONS_V1
+            T Constructor() => (T)context.FunctionBindingContext.CreateObjectInstance(typeof(T), constructorParameters);
+#else
+            T Constructor() => (T)Activator.CreateInstance(typeof(T), constructorParameters);
+#endif
+
+            var state = ((Extensions.DurableTask.DurableEntityContext)context).GetStateWithInjectedDependencies(Constructor);
+
+            object result = method.Invoke(state, args);
+
+            if (method.ReturnType != typeof(void))
+            {
+                if (result is Task task)
+                {
+                    await task;
+
+                    if (task.GetType().IsGenericType)
+                    {
+                        context.Return(task.GetType().GetProperty("Result").GetValue(task));
+                    }
+                }
+                else
+                {
+                    context.Return(result);
+                }
+            }
+        }
+
         void IDurableEntityContext.Return(object result)
         {
             this.ThrowIfInvalidAccess();
             this.CurrentOperationResponse.SetResult(result, this.messageDataConverter);
         }
 
+        internal static MethodInfo FindMethodForContext<T>(IDurableEntityContext context)
+        {
+            var type = typeof(T);
+
+            var interfaces = type.GetInterfaces();
+            const BindingFlags bindingFlags = BindingFlags.IgnoreCase | BindingFlags.Public | BindingFlags.NonPublic | BindingFlags.Instance;
+
+            var method = type.GetMethod(context.OperationName, bindingFlags);
+            if (interfaces.Length == 0 || method != null)
+            {
+                return method;
+            }
+
+            return interfaces.Select(i => i.GetMethod(context.OperationName, bindingFlags)).FirstOrDefault(m => m != null);
```

Comment:
So this gets us the first result of a potential list of matching methods, right? under what conditions would we see a  result list with more than 1 element? Is that even possible for our use-case? 

I think it would be good to add a comment answering the questions above :)

Reply:
I think this is a bug/edge case in the code, and I am a bit hesitant to tackle it in the scope of this PR.

This code assumes that the interfaces customers code against have no method overloads. However, I think we could theoretically support an interface like

```csharp
interface TestInterface 
{
   void TestMethod();

   void TestMethod(string parameter1);
}
```

We could theoretically look at the types of the argument provided to us in the case where the customer has multiple overloads for a single operation, and use that to determine which method overload to call. This does strike me as adding a bit more complicated logic for little gain, but the reflection would not be that challenging.

If we decide this is a scenario we don't want to support, then we should just throw and exception as soon as we detect multiple overloads for a method on an interface, instead of just picking the first, as that would lead to some strange exceptions.

Reply:
@sebastianburckhardt , thoughts?

Reply:
I see what you mean. I'm ok with moving this to a separate follow-up issue and unblocking this PR

Reply:
Interesting case. If entity methods are meant to be different operations, it doesn't make sense to me to have more than one method with the same name. In the switch version of an entity this wouldn't be possible. I don't think we documented, though, that this was a violation. I think it makes sense to enforce this constraint, personally. If we do, I could add an analyzer for this as well.

Reply:
I think it technically _is_ possible in the `switch` entities, you would just have a secondary set of `if-else` statements based on what arguments are present on the operation.

Now whether that is a good design we should encourage is a completely separate discussion...

Reply:
Tracking this here: https://github.com/Azure/azure-functions-durable-extension/issues/1584.